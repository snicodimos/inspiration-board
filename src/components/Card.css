.card {
  background-color: #F4FF81;

  padding: 1em 0;
  margin: 0.5rem;

  min-height: 250px;
  min-width: 225px;
  flex-basis: 18%;

  border-radius: 5px;

  display: grid;
  grid-template-columns: 1fr 2fr 1fr;
  grid-template-rows: 1fr 2fr 1fr;
  align-items: center;
}

.card__content {
  grid-column-start: 2;
  grid-row-start: 2;

  display: flex;
  flex-direction: column;
  justify-content: space-around;

  text-align: center;
  font-weight: 100;
  overflow: hidden;
}

.card__content-text {
  text-overflow: ellipsis;
  overflow: hidden;
}

.card__content-emoji {
  font-size: 3rem;
}

.card__content-text, .card__content-emoji {
  padding: 0;
  margin: 0;
}

.card__delete {
  grid-column-start: 3;
  grid-row-start: 1;
  justify-self: end;

  font-family: 'Permanent Marker', Helvetica, sans-serif;
}
