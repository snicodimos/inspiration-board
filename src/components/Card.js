import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {

  deleteCardClickHandler = (event) => {
    console.log("The delete button was clicked");

    this.props.deleteCardCallback(this.props.id);
  }
  render() {
    return (
      <div className="card">
        <div>
          {this.props.text}
        </div>
        <div>
          {this.props.emoji}
        </div>
        <button
          onClick={ this.deleteCardClickHandler }>
          Delete
        </button>
      </div>
    )
  }
}

Card.propTypes = {

};

export default Card;
