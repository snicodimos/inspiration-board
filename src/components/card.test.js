import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';


describe('Card', () => {
  test('that it matches an existing carc snapshot', () => {

    const wrapper = shallow( <Card
      text="Some positive message or text"
      emoji="heart_eyes"
      deleteCardCallback={() => {} } />);

    expect(wrapper).toMatchSnapshot();
  });
});
