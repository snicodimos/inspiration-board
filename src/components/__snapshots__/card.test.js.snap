// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing carc snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    deleteCardCallback={[Function]}
    emoji="heart_eyes"
    text="Some positive message or text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          Some positive message or text
        </div>,
        <div>
          heart_eyes
        </div>,
        <button
          onClick={[Function]}
        >
          Delete
        </button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Some positive message or text",
        },
        "ref": null,
        "rendered": "Some positive message or text",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "heart_eyes",
        },
        "ref": null,
        "rendered": "heart_eyes",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Some positive message or text
          </div>,
          <div>
            heart_eyes
          </div>,
          <button
            onClick={[Function]}
          >
            Delete
          </button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some positive message or text",
          },
          "ref": null,
          "rendered": "Some positive message or text",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "heart_eyes",
          },
          "ref": null,
          "rendered": "heart_eyes",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
